### PR TITLE
Add support for textContainerInset to ASTextNode (ala UITextView)

### PR DESCRIPTION
**Pull request by garrettmoon**
Hopefully addresses #2000

### CODE REVIEW

**Comment by hannahmbanana** [comment]
cc' @Yue-Wang-Google

File: AsyncDisplayKit/ASTextNode.mm [hunk 1]
```diff
@@ -321,9 +353,14 @@ - (void)calculatedLayoutDidChange
   
   if (layout != nil) {
     ASDN::MutexLocker l(__instanceLock__);
-    if (CGSizeEqualToSize(_constrainedSize, layout.size) == NO) {
-      _constrainedSize = layout.size;
-      _renderer.constrainedSize = layout.size;
+    CGSize layoutSize = layout.size;
+    //textContainerInset
```
Comment (maicki) [inline AsyncDisplayKit/ASTextNode.mm]: Should we change the comment with a bit more information or just remove it. Comment seems kind of half done :)

File: AsyncDisplayKit/ASTextNode.mm [hunk 2]
```diff
@@ -321,9 +353,14 @@ - (void)calculatedLayoutDidChange
...
+    layoutSize.width -= _textContainerInset.left + _textContainerInset.right;
+    layoutSize.height -= _textContainerInset.top + _textContainerInset.bottom;
```
Comment (maicki) [inline AsyncDisplayKit/ASTextNode.mm]: Any opinions adding `()` to `_textContainerInset.left + _textContainerInset.right` just to make it more readable / clear ...

File: AsyncDisplayKit/ASTextNode.mm [hunk 3]
```diff
@@ -335,6 +372,10 @@ - (CGSize)calculateSizeThatFits:(CGSize)constrainedSize
   
   ASDN::MutexLocker l(__instanceLock__);
   
+  //remove textContainerInset
+  constrainedSize.width -= _textContainerInset.left + _textContainerInset.right;
```
Comment (maicki) [inline AsyncDisplayKit/ASTextNode.mm]: Same here maybe adding `()`?

File: AsyncDisplayKit/ASTextNode.mm [hunk 4]
```diff
@@ -353,6 +394,11 @@ - (CGSize)calculateSizeThatFits:(CGSize)constrainedSize
       self.descender *= _renderer.currentScaleFactor;
     }
   }
+  
+  //add textContainerInset
+  size.width += _textContainerInset.left + _textContainerInset.right;
```
Comment (maicki) [inline AsyncDisplayKit/ASTextNode.mm]: Same here maybe adding ()?

**Comment by Adlai-Holler** [comment]
@garrettmoon Sweet! Would you be willing to add a snapshot test case to watch for regressions in this system?

**Comment by Yue-Wang-Google** [comment]
Thanks a lot for the new feature! We'll modify our code to utilize this as a solution to solve the italic text clipping bug.

**Comment by ocrickard** [comment]
Removing https://github.com/facebook/AsyncDisplayKit/blob/aba05a747c9222de91de2a65e601393a7a284943/AsyncDisplayKit/TextKit/ASTextKitContext.mm#L56  may help with the italic issue.

**Comment by ocrickard** [comment]
I'm in support of this diff. It's ultimately what Apple decided to do with UITextView.

**Comment by Adlai-Holler** [comment]
Thanks @garrettmoon this is killer!

**Pull request merged**